src/wasm-interpreter.h

```diff
@@ -902,286 +909,369 @@ class ModuleInstanceBase {
   // stack traces.
   std::vector<Name> functionStack;
 
-public:
-  // Call a function, starting an invocation.
-  Literal callFunction(Name name, const LiteralList& arguments) {
-    // if the last call ended in a jump up the stack, it might have left stuff for us to clean up here
-    callDepth = 0;
-    functionStack.clear();
-    return callFunctionInternal(name, arguments);
+  std::unordered_set<size_t> droppedSegments;
+
+  void initializeTableContents() {
+    for (auto& segment : wasm.table.segments) {
+      Address offset = (uint32_t)ConstantExpressionRunner<GlobalManager>(globals).visit(segment.offset).value.geti32();
+      if (offset + segment.data.size() > wasm.table.initial) {
+        externalInterface->trap("invalid offset when initializing table");
+      }
+      for (size_t i = 0; i != segment.data.size(); ++i) {
+        externalInterface->tableStore(offset + i, segment.data[i]);
+      }
+    }
   }
 
-  // Internal function call. Must be public so that callTable implementations can use it (refactor?)
-  Literal callFunctionInternal(Name name, const LiteralList& arguments) {
+  void initializeMemoryContents() {
+    Builder builder(wasm);
 
-    class FunctionScope {
```

Comment:
`FunctionScope` and `RuntimeExpressionRunner` were moved out of `ModuleInstanceBase::callFunction` be be private inner classes of `ModuleInstanceBase`. This allows `ModuleInstanceBase`'s constructor to use the `RuntimeExpressionRunner` to visit memory initialization functions it synthesizes. This generated a big diff because the indentation for all this code changed. Doing `git diff --patience -w` on the command line gives a much more sane diff.